clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2683,6 +2702,14 @@ private void rescheduleFindCoordinatorTask(ConsumerGroupOperationContext<?, ?> c
                 context.node().orElse(null));
         // Requeue the task so that we can try with new coordinator
         context.setNode(null);
+
+        Call call = nextCall.get();
+
+        if (call.tries > maxRetries) {
```

Comment:
I missed this on the first pass, but would there be any harm moving this call to `enqueue`? That way we do not need to repeat this check for other scenarios involving chaining or partial failures. (For example, I assume we'd end up doing the same thing in `rescheduleMetadataTask`).

We should probably also try to reuse `failWithTimeout` for a consistent exception message.

Reply:
I can't think of any issues with moving this logic to `enqueue`, the only reason I put this inside the `rescheduleFindCoordinatorTask` was so that we could avoid making a call to find the coordinator if the call had maxed out on retries. I will try it with this logic to see if there are any issues with moving it to `enqueue`.

Also the `fail` method already calls `failWithTimeout` when a call has maxed out the retries: https://github.com/apache/kafka/blob/5a17d56fca676c70579222b1fbe5d1dd501c9297/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L757-L759